leapp/cli/report/__init__.py

```diff
@@ -0,0 +1,79 @@
+"""
+This module is currently single workflow specific (upgrades). After introducing other use cases we can make it
+more robust
+"""
+
+import os
+
+from leapp.exceptions import CommandError
+from leapp.utils.clicmd import command, command_opt, command_arg
+from leapp.utils.report import fetch_upgrade_report_messages
+from leapp.cli.upgrade import fetch_last_upgrade_context
+
+from jinja2 import Template
+import json
+
+
+@command('report', help='Create report for last upgrade execution (default) or specific execution id using --id ID')
+@command_opt('id')
+@command_opt('--format')
+def report(args):
+    if os.getuid():
+        raise CommandError('This command has to be run under the root user.')
+
+    id = args.id or fetch_last_upgrade_context()[0]
+
+    if not id:
+        raise CommandError('No previous Leapp upgrade run of the requested context {} found'.format(id))
+
+    messages = fetch_upgrade_report_messages(id)
+    if not messages:
+        raise CommandError('No upgrade report messages found for context {}'.format(id))
+
+    # FIXME: add proper default and choices parametrs for the format option
+    if args.format not in ('html', 'plaintext'):
+        print('No / not supported renderer provided as --format (available renderers: "html", "plaintext"). '
+              'Defaulting to "plaintext"')
+        print('\n')
+        renderer = 'plaintext'
+    else:
+        renderer = args.format
+
+    messages_to_print = []
+    for message in messages:
+        topic = message.get('topic')
+        phase = message.get('phase')
+        data = json.loads(message.get('message').get('data'))
+        merged_data = {'topic': topic, 'phase': phase}
+
+        for key in ('detail', 'named'):
+            if key in merged_data:
+                raise ValueError('Duplicate key {} in report structure'.format(key))
+            merged_data[key] = json.loads(data.get(key))
+
+        if not merged_data['named']:
+            merged_data['named'] = {}
+
+        if 'title' in merged_data['named'] or 'title' in merged_data['detail']:
```

Comment:
I don't know but this feels to be the wrong place to check for this. We should rather think of additional validators for the models instead of making a check like this here.